alerting/src/test/kotlin/com/amazon/opendistroforelasticsearch/alerting/MonitorRunnerIT.kt

```diff
@@ -507,6 +512,97 @@ class MonitorRunnerIT : AlertingRestTestCase() {
+
+        assertEquals(actionResults1.size, 2)
+
+        assertEquals(actionResults1[actionThrottleEnabled.id]!!.lastExecutionTime,
+                actionResults2[actionThrottleEnabled.id]!!.lastExecutionTime)
+    }
+
+    fun `test monitor with throttled action for different alerts`() {
+        val actionThrottleEnabled = randomAction(template = randomTemplateScript("Hello {{ctx.monitor.name}}"),
+                destinationId = createDestination().id,
+                throttleEnabled = true, throttle = Throttle(value = 5, unit = MINUTES))
+        val actions = listOf(actionThrottleEnabled)
+        val trigger = randomTrigger(condition = ALWAYS_RUN, actions = actions)
+        val monitor = createMonitor(randomMonitor(triggers = listOf(trigger),
+                schedule = IntervalSchedule(interval = 1, unit = ChronoUnit.MINUTES)))
+
+        verifyActionThrottleResults(monitor.id, mutableMapOf(Pair(actionThrottleEnabled.id, false)))
+
+        val alerts1 = searchAlerts(monitor)
+        assertEquals("1 alert should be returned", 1, alerts1.size)
+        verifyAlert(alerts1.single(), monitor, ACTIVE)
+        val actionResults1 = verifyActionExecutionResultInAlert(alerts1[0], mutableMapOf(Pair(actionThrottleEnabled.id, 0)))
+
+        updateMonitor(monitor.copy(triggers = listOf(trigger.copy(condition = NEVER_RUN)), id = monitor.id))
+        executeMonitor(monitor.id)
+        val alerts2 = searchAlerts(monitor, AlertIndices.ALL_INDEX_PATTERN).single()
+        verifyAlert(alerts2, monitor, COMPLETED)
+
+        updateMonitor(monitor.copy(triggers = listOf(trigger.copy(condition = ALWAYS_RUN)), id = monitor.id))
+        verifyActionThrottleResults(monitor.id, mutableMapOf(Pair(actionThrottleEnabled.id, false)))
+        val alerts3 = searchAlerts(monitor)
+        assertEquals("1 alert should be returned", 1, alerts1.size)
+        verifyAlert(alerts1.single(), monitor, ACTIVE)
+        assertNotEquals(alerts1[0].id, alerts3[0].id)
+
+        val actionResults3 = verifyActionExecutionResultInAlert(alerts3[0], mutableMapOf(Pair(actionThrottleEnabled.id, 0)))
+
+        assertNotEquals(actionResults1[actionThrottleEnabled.id]!!.lastExecutionTime,
+                actionResults3[actionThrottleEnabled.id]!!.lastExecutionTime)
+    }
+
+    private fun verifyActionExecutionResultInAlert(alert: Alert, expectedResult: Map<String, Int>):
+            MutableMap<String, ActionExecutionResult> {
+        val actionResult = mutableMapOf<String, ActionExecutionResult>()
+        for (result in alert.actionExecutionResults) {
+            val expected = expectedResult[result.actionId]
+            assertEquals(expected, result.throttledCount)
+            actionResult.put(result.actionId, result)
+        }
+        return actionResult
+    }
+
+    private fun verifyActionThrottleResults(monitoId: String, expectedResult: Map<String, Boolean>) {
+        val output = entityAsMap(executeMonitor(monitoId))
```

Comment:
Could you move this where this function is being called. Because name says verify but it also executeMonitor.